### ISSUE TITLE
Global functions are not properly counted

### ISSUE THREAD

**Issue by Seb35** [sebastianbergmann/php-code-coverage#468]
I found two issues – I guess bugs – with global functions, mainly in HTML and XML reports (it seems this metric is not reported in other formats – I’m quite new in the world of testing).
- The totals of global functions are always 0/0 even if they individually have the score 1/1, in the file itself and in the parent directory.
- The CRAP is always 0 for global functions; given they are not no-op, the CRAP should be 1 or greater.

I tested with PHPUnit 5.5.2 + PHP_CodeCoverage 4.0.1 with the three drivers: PHP 5.6.24+Xdebug 2.2.5, PHP 7.0.9+Xdebug 2.4.0, PHPDBG 7.0.9. They all report the same figures. Below are the screenshot
![bug_php_codecoverage_directory](https://cloud.githubusercontent.com/assets/772681/17823648/da62cb6e-665d-11e6-93c7-17f98683ac0a.png)
![bug_php_codecoverage_file](https://cloud.githubusercontent.com/assets/772681/17823655/deeda7e4-665d-11e6-9284-5e8a1a5ef9ff.png)

The XML returns, between other things:
`<functions count="1" tested="0" percent="0.00%"/>`
`<function name="a" signature="a($b, $c=false)" start="3" crap="0" executable="3" executed="3" coverage="0"/>`

I searched around existing open and closed issues, I didn’t find such issue. Other have similar effects (total = 0/0) because of no executable lines (#428) or other issues but unrelated to global functions (#422).

**Comment by Seb35** [comment]
I investigated and found the CRAP and coverage (percentage) are not updated in src/Node/File.php, at the contrary of methods. This solved the issue with CRAP and coverage in HTML and XML.

For the totals, only the methods are counted. Similarly to the statistics "Classes and Traits", the statistics "Functions and Methods" should be the addition of the two figures.

I prepared a patch, but I don’t have tests for now. I saw the existing tests only have an empty global function, so the CRAP is really 0 in this case (and its coverage is the issue mentionned elsewhere with 0/0). And the totals are only visible in the XML format – in addition of HTML – right? XML would be more testable than HTML.

**Comment by Seb35** [comment]
The patch I proposed is failing two tests, the same in HTML and XML. The global function tested (which I didn’t remarked previously in tests) only has one operation with an annotation //@codeCoverageIgnore so I don’t know if it should be considered as “a tested function with no testable operations” or “an untested function since there is no testable operations”. Probably other tests with no-empty global functions should be added.

**Comment by sebastianbergmann** [comment]
Thank you for your report.

Please provide a **minimal**, **self-contained**, reproducing test case that shows the problem you are reporting.

Without such a minimal, self-contained, reproducing test case I will not be able to investigate this issue.

**Comment by direvus** [comment]
The minimal test case is trivial.  Just write any global function, write a test that covers the global function, and then produce a coverage report for it.

For example:

src/foo.php:
```
<?php
function foo(){
    return 'foo';
}
?>
```

tests/FooTest.php:
```
<?php
use \PHPUnit\Framework\TestCase;

require_once('src/foo.php');

class FooTest extends TestCase {
    /**
     * @covers ::foo
     */
    public function testFoo(){
        $this->assertSame('foo', foo());
    }
}
?>
```

Then run the coverage report:

`$ phpunit --whitelist src --coverage-html=coverage tests`

I would expect to see 1/1 coverage under "Functions and Methods" but instead I see 0/0.  The "Lines" metric correctly reports 1/1 lines of code covered.

Confirmed against PHPUnit 5.7.17 + XDebug 2.4.1.

**Comment by sebastianbergmann** [comment]
Fixed by c0f32b5047b7a7a1b7698a9118018c09c24c972a.
